### PR TITLE
components: document using of event parameters in $emit

### PR DESCRIPTION
**Pull request by dennybiasiolli**
ref. #1336

### CODE REVIEW

File: src/v2/guide/components.md [hunk 1]
```diff
@@ -507,7 +507,7 @@ We have learned that the parent can pass data down to the child using props, but
 Every Vue instance implements an [events interface](../api/#Instance-Methods-Events), which means it can:
 
 - Listen to an event using `$on(eventName)`
-- Trigger an event using `$emit(eventName)`
+- Trigger an event using `$emit(eventName[, optionalPayload])`
```
Comment (sdras) [inline src/v2/guide/components.md]: This should be `$emit(eventName, [optionalPayload])`

Comment (sdras) [reply src/v2/guide/components.md]: Though, truthfully, it doesn't need to be in an array, so maybe that should be more clear, too.

**Comment by sdras** [comment]
I really like this addition! This is definitely something we should be covering a bit better in the docs. I made some comments, if we resolve those we'll be close to merging.

File: src/v2/guide/components.md (hunk 1 above)
Comment (dennybiasiolli) [reply src/v2/guide/components.md]: @sdras I put it into square brackets to specify it is optional, do you have any suggestion on the format?
If you think it's better, I can write it in this way
>  Trigger an event using `$emit(eventName)` or `$emit(eventName, optionalPayload)`
In order to make it explicit ;)

Comment (dennybiasiolli) [reply src/v2/guide/components.md]: Or I can edit as
> Trigger an event using `$emit(eventName, optionalPayload)`

if it's clear enough

**Comment by sdras** [comment]
LGTM 👍

**Pull request merged**